### PR TITLE
Add import-ap validations to sar.sch

### LINKED ISSUE #506: Add `import-ap` validations to `sar.sch`
**Extended Description**
- Add `import-ap` validations to `sar.sch`. Note that non-OSCAL SAPs are **not allowed**.

**Acceptance Criteria**
- [x] All Schematron assertion messages are declarative statements which affirm the positive test outcome.
- [x] All Schematron assertion diagnostic messages are declarative statements which explain the negative test outcome.
- [x] The Schematron code has no assertion failures when validated using `src/validations/styleguides/sch.sch` using the `basic` phase.
- [x] XSpec unit tests for positive and negative Schematron assertion outcomes accompany all Schematron assertions (where feasible).
- Story tasks completed.

**Story Tasks**
- [x] Add `import-ap` validations to `sar.sch`. Use the `import-ssp` in `sap.sch` for analogous validations.

**Definition of Done**
 - [x] Acceptance criteria met
 - [x] Unit test coverage of our code > 95%  
 - [x] Automated code quality checks passed 
 - [x] Security reviewed and reported
 - [x] Reviewed against plain language guidelines
 - [x] Code must be self-documenting
 - [x] No local tech debt
 - [x] Load/performance tests passed
 - [x] Documentation updated
 - [x] Architectural Decision Record completed as necessary for significant design choices
 - [ ] PR reviewed & approved
 - [ ] Source code merged

### PR DESCRIPTION
**Pull request by GaryGapinski**
Closes #506.

### CODE REVIEW

File: src/validations/test/rules/sar.xspec [hunk 1]
```diff
@@ -2,35 +2,305 @@
...
+                label="when the import-ap element">
+                <x:scenario
+                    label="is present">
+                    <x:expect-not-assert
+                        id="has-import-ap"
+                        label="that is correct" />
+                </x:scenario>
+            </x:scenario>
+            <x:scenario
+                label="when the import-ap element">
```
Comment (danielnaab) [inline src/validations/test/rules/sar.xspec:43]: This seems superfluous.

```suggestion
```

File: src/validations/test/rules/sar.xspec [hunk 2]
```diff
@@ -2,35 +2,305 @@
...
+                        </assessment-plan>
+                    </x:context>
+                    <x:expect-assert
+                        id="has-import-ap"
+                        label="that is an error" />
+                </x:scenario>
+            </x:scenario>
+
+            <x:scenario
+                label="when the import-ap">
```
Comment (danielnaab) [inline src/validations/test/rules/sar.xspec:59]: Ditto on this an the next one - do we have a convention on when we nest scenarios?

**Review by danielnaab — APPROVED** [review]
lgtm. I don't have specific comments on the rules, just wondering on when scenarios are to be nested.

**Review by markXLIX — APPROVED** [review]
Looks good

File: src/validations/test/rules/sar.xspec (hunk 2 above)
Comment (GaryGapinski) [reply src/validations/test/rules/sar.xspec:59]: It is superfluous.

File: src/validations/test/rules/sar.xspec (hunk 1 above)
Comment (GaryGapinski) [reply src/validations/test/rules/sar.xspec:43]: It is superfluous.

**Pull request merged**